### PR TITLE
Show CCI attribute data_type as type_of_data

### PR DESCRIPTION
**Pull request by TonioF**
Addresses https://github.com/dcs4cop/xcube-cci/issues/54. The changes in this PR make it possible again to search for data sets based on their data_type/type_of_data. Also, the term type_of_data is used in the dataset descriptors, preventing any possible further confusion (as, e.g., in the metadata display in the CCI toolbox).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ## Changes in 0.9.1 (in development)
 * Fixed issue that datasets with spatial bounds could not be opened.
 * Show version as `__version__`
+* The CCI data attribute `data_type` has been renamed to `type_of_data` to
```
Comment (forman) [inline CHANGES.md:4]: What is a "CCI data attribute" and where is it used? What is the impact for users?

File: xcube_cci/dataaccess.py [hunk 2]
```diff
@@ -424,7 +426,7 @@ def get_search_params_schema(cls, data_type: str = None) -> JsonObjectSchema:
                           .replace('-yrs', ' years').replace('yr', 'year')
                            for data_id in data_ids])
         processing_levels = set([data_id.split('.')[3] for data_id in data_ids])
-        data_types = set([data_id.split('.')[4] for data_id in data_ids])
+        types_of_data = set([data_id.split('.')[4] for data_id in data_ids])
```
Comment (forman) [inline xcube_cci/dataaccess.py:429]: A much cleaner approach would be to have a search parameter `attrs` that may include search attributes and where the attribute  `data_type` is included as specified by the ODP. 

The names `data_type` and `types_of_data ` side-by-side as search parameters is hard to explain to users (and to me).

**Review by forman — CHANGES_REQUESTED** [review]
Please check if search params can be changed to the following so they are more in line with dataset descriptors:

- `time_range`
- `bbox`
- `crs`                    # to search for datasets that are not in WGS84
- `data_type`         # to filter gridded data or shapefiles
- `attrs`                  #  would comprise all that ODP-specific "facets" inclusing `data_type`

File: CHANGES.md (hunk 1 above)
Comment (TonioF) [reply CHANGES.md:4]: I use this term to refer to attributes of CCI data. In the CCI context, there are certain attributes that many or all datasets have. In particular, a cci dataset id is built from several of these attributes. The term 'dataType' is also used as an opensearch parameter.

File: xcube_cci/dataaccess.py (hunk 2 above)
Comment (AliceBalfanz) [reply xcube_cci/dataaccess.py:429]: I agree with forman :) would prefer a search param including attrs

**Review by forman — APPROVED** [review]
Thanks!

**Pull request merged**